### ISSUE TITLE
Browser back button behaviour

### ISSUE THREAD

**Issue by mark-prins** [msupply-foundation/open-msupply#4907]
## What went wrong? 😲

With the current workflow
* I'm on page (x)
* I navigate to the Internal orders list
* Click on 'new order'
* the newly created internal order loads
* click the back button in the browser
* you're on page (x)

## Expected behaviour 🤔
You should be back on the internal order list
Same problem with 
* Inbound shipment
* Outbound shipment
* Prescription
* Supplier return

## How to Reproduce 🔨

Steps to reproduce the behaviour:

[ see above ]

## Your environment 🌱

- Open mSupply Version:
- Legacy mSupply Central Server Version:

- Platform:

- Database type:

**Comment by mark-prins** [comment]
In `RequestRequisition/ListView/AppBarButtons` there is `{ replace: true }` on the navigate call which happens after creation. This should be false, which I think is the default.

**Comment by aimee-mcneil-melville** [comment]
List views updated:

- Inbound Shipment
- Outbound Shipment
- Prescriptions
- Request Requisition / Internal Orders
- Response Requisition / Requisitions

Added navigate to Supplier Return

**Comment by marthakeezy** [comment]
Hi team, 

@lache-melvin @aimee-mcneil-melville @mark-prins 

Could someone please update the testing steps with the expected behaviour when it comes to the new full-screen editing modals? Currently, once let's say a Requisition is saved, back button does not take you to the list view, but instead takes you back through the entire process of adding an item. Is this to be expected?

I feel like a user can get stuck in an infinite loop of adding items and navigating back only to be prompted to add more haha

Video attached ->>> [here](https://drive.google.com/drive/folders/1-ZpLRCS_DAy2kLkmpMz_Rx3LjJYkQWZF)

**Comment by lache-melvin** [comment]
Hm good question - I think the item selection being on the edit page got introduced after this issue closed!

While I think this is working as could be "expected" (clicking back takes you to whatever URL you were on before, including the `/new` one) I agree, we could remove the "/new" page from the browser history, so we skip over it if we were to click back.

I.e. Detail view ->  [click add item] -> new item view on edit page -> [select an item] -> edit item view on edit page -> [click back] -> detail view

**Comment by Chris-Petty** [comment]
@marthakeezy is this going to be picked up for retesting even when it's open or should we close it?

**Comment by marthakeezy** [comment]
@Chris-Petty i don't think this has been looked at by anyone, right? because i don't see any new PRs and i think it's still in 'Backlog' in omSupply sprint? usually when the dev fixes it. they close it? correct if wrong :)

**Comment by Chris-Petty** [comment]
@marthakeezy yup you're right. ngl I half the updates were hidden so I didn't see and may have assumed dev was already done for some reason.

**Comment by marthakeezy** [comment]
haha no worries! thanks for checking :)

**Comment by marthakeezy** [comment]
# 🧪 Testing



`V2.6.1-RC4`

- [ ] Go to Distribuaiton -> Requisitions
- [ ] Create of edit a requisition
- [ ] Click `Add Item`
- [ ] Choose an item
- [ ] Enter quantity/save
- [ ] Close the edit view
- [ ] Use the browser back button it should go back to the item edit view but to an existing item - not the new item
- [ ] Repeat with adding an item from the edit list view, and with other pages (internal orders, prescriptions)
